src/ctia/http/routes/crud.clj

```diff
@@ -416,7 +416,7 @@
                         (store/delete-search
                           query
                           identity-map
-                          (wait_for->refresh (:wait_for params))))
```

Comment:
I think you still want to also set `:refresh true` in case `:wait_for` is true. Otherwise, search requests still might get deleted data during some time after delete task is finished. I stumbled upon this behaviour during testing cleanup logic for iroh-events.

Reply:
that's a good suggestion, but it will force a full refresh of the index, and I am worried that it might be an attack vector if we enabled that.

Reply:
true, it is possible to degrade performance of a cluster by overusing `refresh=true`. At the same time, I feel safe if we know for sure when this method is called and don't expose it for direct http call. :)

Reply:
> At the same time, I feel safe if we know for sure when this method is called and don't expose it for direct http call. :)

Here that's the parameter exposed in all CRUD route that is transformed in "wait_for", `wait_for->refresh` never sets the parameter to "true". We could introduce the use of "true" later if we find a safe scenario of the warning from ES doc:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html
"This should ONLY be done after careful thought and verification that it does not lead to poor performance, both from an indexing and a search standpoint."
"true creates less efficient indexes constructs (tiny segments) that must later be merged into more efficient index constructs (larger segments). Meaning that the cost of true is paid at index time to create the tiny segment, at search time to search the tiny segment, and at merge time to make the larger segments. "